src/AWS.Deploy.Orchestration/DeploymentSettingsHandler.cs

```diff
@@ -91,5 +100,53 @@ public async Task ApplySettings(DeploymentSettings deploymentSettings, Recommend
 
             throw new InvalidDeploymentSettingsException(DeployToolErrorCode.DeploymentConfigurationNeedsAdjusting, errorMessage.Trim());
         }
+
+        public async Task<string> SaveSettings(string filePath, Recommendation recommendation, CloudApplication cloudApplication, OrchestratorSession orchestratorSession)
+        {
+            var parentDirectory = Path.GetDirectoryName(filePath);
+            if (string.IsNullOrEmpty(parentDirectory))
+            {
+                return $"Failed to save deployment settings because {filePath} is not a valid path";
+            }
+            if (!_directoryManager.Exists(parentDirectory))
+            {
+                return $"Failed to save deployment settings because part of {filePath} does not exist on disk";
+            }
+
+            var deploymentSettings = new DeploymentSettings
+            {
+                AWSProfile = orchestratorSession.AWSProfileName,
+                AWSRegion = orchestratorSession.AWSRegion,
+                ApplicationName = cloudApplication.Name,
+                RecipeId = cloudApplication.RecipeId,
+                Settings = new Dictionary<string, object>()
+            };
+
+            var optionSettings = recommendation.GetConfigurableOptionSettingItems();
+            foreach (var optionSetting in optionSettings)
+            {
+                var id = optionSetting.FullyQualifiedId;
+                var value = _optionSettingHandler.GetOptionSettingValue(recommendation, optionSetting);
+                deploymentSettings.Settings[id] = value;
+            }
+
+            try
+            {
+                var content = JsonConvert.SerializeObject(deploymentSettings, new JsonSerializerSettings
+                {
+                    Formatting = Formatting.Indented,
+                    NullValueHandling = NullValueHandling.Ignore,
+                    ContractResolver = new SerializeModelContractResolver()
+                });
+
+                await _fileManager.WriteAllTextAsync(filePath, content);
+            }
+            catch (Exception ex)
+            {
+                return $"Failed to save the deployment settings at {filePath} due to the following error: {Environment.NewLine}{ex.Message}";
```

Comment:
An expected exception should be thrown here. Also make sure to pass the generic exception ex to the inner exception of the expected one.

Reply:
Done